runtime/js.rs

```diff
@@ -1,14 +1,33 @@
 // Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
 use deno_core::Snapshot;
 use log::debug;
+use once_cell::sync::Lazy;
 
-pub static CLI_SNAPSHOT: &[u8] =
-  include_bytes!(concat!(env!("OUT_DIR"), "/CLI_SNAPSHOT.bin"));
+pub static CLI_SNAPSHOT: Lazy<Box<[u8]>> = Lazy::new(
+  #[cold]
+  #[inline(never)]
+  || {
+    static COMPRESSED_CLI_SNAPSHOT: &[u8] =
+      include_bytes!(concat!(env!("OUT_DIR"), "/CLI_SNAPSHOT.bin"));
+
+    let size =
+      u32::from_le_bytes(COMPRESSED_CLI_SNAPSHOT[0..4].try_into().unwrap())
+        as usize;
+    let mut vec = Vec::with_capacity(size);
+
+    unsafe {
+      vec.set_len(size);
+    }
```

Comment:
```suggestion
    let mut vec = vec![0; size];
```
Why not just this? I don't like unsafe :-)

Reply:
I would much prefer that tiny perf hit, than an `unsafe` block without a `// SAFETY:` comment. I am surprised the linter isn't having a mental breakdown because of that anyway.